include/h2o/socketpool.h

```diff
@@ -176,7 +180,7 @@ void h2o_socketpool_unregister_loop(h2o_socketpool_t *pool, h2o_loop_t *loop);
  * connects to the peer (or returns a pooled connection)
  */
 void h2o_socketpool_connect(h2o_socketpool_connect_request_t **_req, h2o_socketpool_t *pool, h2o_url_t *url, h2o_loop_t *loop,
-                            h2o_multithread_receiver_t *getaddr_receiver, h2o_socketpool_connect_cb cb, void *data);
+                            h2o_multithread_receiver_t *getaddr_receiver, h2o_iovec_t alpn_protos, h2o_socketpool_connect_cb cb, void *data);
```

Comment:
I think we have to select h1 or h2 per request due to load balancing between those. Do you mean we 

> because a socketpool is expected to contain a set of sockets that are used for certain protocol.

If I recall collectly, in our previous discussion we agreed that we use one socketpool for both getting h1 and h2 sockets, doesn't it?

Reply:
> If I recall collectly, in our previous discussion we agreed that we use one socketpool for both getting h1 and h2 sockets

Ah. You are correct. Thank you for pointing that out.

Then, could we have a doc-comment describing how the parameter should be used? I'd assume that the proper behavior when passing multiple protocol names is that you should only return connections that share "one" protocol name (e.g. h1) and close the connections that uses any other negotiated protocol.